tests/als_test.py

```diff
@@ -81,6 +82,32 @@ def test_factorize(self):
                                                        % (i, j, reconstructed[i, j], dtype, use_cg,
                                                           use_native))
 
+    def test_explain(self):
+        counts = csr_matrix([[1, 1, 0, 1, 0, 0],
+                             [0, 1, 1, 1, 0, 0],
+                             [1, 0, 1, 0, 0, 0],
+                             [1, 1, 0, 0, 0, 0],
+                             [0, 0, 1, 1, 0, 1],
+                             [0, 1, 0, 0, 0, 1],
+                             [0, 0, 0, 0, 1, 1]], dtype=np.float64)
+        user_items = counts * 2
+
+        model = AlternatingLeastSquares(factors=4,
+                                        regularization=1e-1,
+                                        use_native=False,
+                                        use_cg=False,
+                                        iterations=10)
+        np.random.seed(23)
+        model.fit(user_items)
+        model.item_factors, model.user_factors
+
+        userid = 0
+        recs = model.recommend(userid, user_items, N=10, recalculate_user=True)
```

Comment:
I'm not sure why the test fails if we set `recalculate_user=False`

Looks like the user factors are not really converging. I can try to debug this further tomorrow.